allennlp/training/metrics/evalb_bracketing_scorer.py

```diff
@@ -0,0 +1,127 @@
+
+from typing import List
+import os
+import tempfile
+import re
+import math
+import subprocess
+
+from overrides import overrides
+from nltk import Tree
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.training.metrics.metric import Metric
+
+
+@Metric.register("evalb")
+class EvalbBracketingScorer(Metric):
+    """
+    This class uses the external EVALB software for computing a broad range of metrics
+    on parse trees. Here, we use it to compute the Precision, Recall and F1 metrics.
+    You can download the source for EVALB from here: <http://nlp.cs.nyu.edu/evalb/>.
```

Comment:
You should note somewhere in here that this is calling a subprocess and writing and reading files from disk, so you _definitely_ don't want to compute this in the inner loop of your training code.